components/director/internal/domain/formation/service.go

```diff
@@ -163,18 +194,18 @@ func (s *service) CreateFormation(ctx context.Context, tnt string, formation mod
 	}
 
 	// TODO:: Currently we need to support both mechanisms of formation creation/deletion(through label definitions and Formations entity) for backwards compatibility
-	if err := s.createFormation(ctx, tnt, templateName, formationName); err != nil {
-		return nil, err
-	}
-
-	return &model.Formation{Name: formationName}, nil
+	return s.createFormation(ctx, tnt, templateName, formationName)
 }
 
 // DeleteFormation removes the provided formation from the scenario label definitions of the given tenant.
 // Also, removes the Formation entity from the DB
 func (s *service) DeleteFormation(ctx context.Context, tnt string, formation model.Formation) (*model.Formation, error) {
 	formationName := formation.Name
-	err := s.modifyFormations(ctx, tnt, formationName, deleteFormation)
+	if err := s.modifyFormations(ctx, tnt, formationName, deleteFormation); err != nil {
+		return nil, err
+	}
+
+	f, err := s.getFormationByName(ctx, formation.Name, tnt)
```

Comment:
Because only the name is available. You can see that in the schema, `FormationInput` don't have the formation ID. And we want all of the functions for formations(create, delete, assign, unassign) to return the whole formation as a result, so we need to get the formation from the DB and the only property that we have available is the formation name.